src/preload.ts

```diff
@@ -87,6 +92,47 @@ class Kiosk implements KioskBrowser.Kiosk {
     return ipcRenderer.invoke(unmountUsbDriveChannel, device)
   }
 
+  public async getFileSystemEntries(path: string): Promise<FileSystemEntry[]> {
+    debug('forwarding `getFileSystemEntries` to main process')
+    const result: FileSystemEntry[] = await ipcRenderer.invoke(
+      fileSystemGetEntriesChannel,
+      path,
+    )
+    return result.map(entry => ({
+      ...entry,
+      mtime:
+        typeof entry.mtime === 'string' ? new Date(entry.mtime) : entry.mtime,
+      atime:
+        typeof entry.atime === 'string' ? new Date(entry.atime) : entry.atime,
+      ctime:
+        typeof entry.ctime === 'string' ? new Date(entry.ctime) : entry.ctime,
+    }))
+  }
+
+  public async readFile(path: string): Promise<Buffer>
+  public async readFile(path: string, encoding: string): Promise<string>
+  public async readFile(...args: unknown[]): Promise<Buffer | string> {
+    debug('forwarding `readFile` to main process')
+    return ipcRenderer.invoke(fileSystemReadFileChannel, ...args)
+  }
+
+  public async writeFile(path: string): Promise<FileWriter>
+  public async writeFile(path: string, content: Buffer | string): Promise<void>
+  public async writeFile(
+    path: string,
+    content?: Buffer | string,
+  ): Promise<FileWriter | void> {
+    debug('forwarding `writeFile` to main process')
+    const writer = await FileWriter.fromPath(path)
+
+    if (typeof content !== 'undefined') {
+      await writer.write(content)
+      await writer.end()
+    } else {
+      return writer
+    }
```

Comment:
what's the reason for having the 3 fine-grained underlying api calls (open, write, end) vs just one, since we're not exposing streaming anyways? That potentially opens up some attack surface, i.e. can I write to any file descriptor from web content now? I would suggest having the IPC interface be just one single API call with the content and the path, no exposing of file descriptors.

Reply:
We don't expose the streaming API here, but we do with `saveAs` and this is built on the same thing that is. If we don't do streaming we can't feasibly use this API to save large things like the scanner backup (GBs of data).

Reply:
And you could write to any file descriptor that your origin already opened, but only if you also manage to get the code in the preload context to do the IPC call for you. Not sure how to do that offhand, and it probably requires a vuln.

Reply:
Actually, getting the prototype of the object returned from `saveAs` may allow such access.

Reply:
This should be fixed now in 70eb559.